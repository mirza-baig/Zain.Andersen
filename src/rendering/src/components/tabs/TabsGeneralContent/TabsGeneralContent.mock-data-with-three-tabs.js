const defaultData = {
    uid: "bf184462-f031-48c2-af01-0c1718500ef7",
    componentName: "TabsGeneralContent",
    dataSource: "{B2BB8B58-EC33-401C-852D-17EEE705BC9F}",
    rendering: {
        dataSource: 'sampledatasource',
        componentName: 'TabsGeneralContent',
    },
    fields : {
      componentSpacing: null,
      sectionId: {
        value: ""
      },
      theme: {
        value: ""
      },
      headlineLevel: {
        id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
        url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
        name: "Heading 2",
        displayName: "Heading 2",
        fields: {
          Value: {
            value: "h2"
          }
        },
        templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
        templateName: "Enum"
      },
      headlineText: {
        value: "This is the headline text"
      },
      tabLabelView: {
        id: "8655afb5-cb94-42b0-b942-fe2a5a11a304",
        url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Tab-Labels/Standard-View",
        name: "Standard View",
        displayName: "Standard View",
        fields: {
          Value: {
            value: "long"
          }
        },
        templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
        templateName: "Enum"
      },
      defaultActiveTab: null,
      eventName: {
        value: ""
      },
      eventType: {
        value: ""
      },
      eventZone: {
        value: ""
      },
      children: [
        {
          id: "6bed4bde-b050-4dce-9360-87cbd18da911",
          fields: {
            tabAnchor: {
              value: ""
            },
            headlineLevel: {
              id: "7f10288a-beb1-4d95-a139-32df40094441",
              url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
              name: "Heading 3",
              displayName: "Heading 3",
              fields: {
                Value: {
                  value: "h3"
                }
              },
              templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
              templateName: "Enum"
            },
            headlineText: {
              value: "Tab Item 1 here is some long text blah blah"
            },
          },
          placeholders: {
            components: [
              {
                componentName: "ContentBlockWithMedia1",
                uid: "56d2beff-74da-4aef-9051-061ca573176b",
                dataSource: "{56D2BEFF-74DA-4AEF-9051-061CA573176B}",
                fields: {
                  primaryMediaCaption: {
                    value: "This is a caption with a <a href=\"/\">link</a>."
                  },
                  secondaryMediaCaption: {
                    value: "This is a caption with a <a href=\"/\">link</a>."
                  },
                  primaryMediaItem: {
                    id: "1fbaaf0c-1a06-4d0b-b21f-2d4f2f27f764",
                    url: "/Data/Elements/YouTube-Video",
                    name: "YouTube Video",
                    displayName: "YouTube Video",
                    fields: {
                      youTubeShowControls: {
                        value: true
                      },
                      youTubeAutoLoop: {
                        value: true
                      },
                      youTubeAutoPlay: {
                        value: false
                      },
                      youTubeDisableKeyboard: {
                        value: true
                      },
                      youTubeModestBranding: {
                        value: true
                      },
                      youTubeMute: {
                        value: false
                      },
                      videoHeight: {
                        value: 338
                      },
                      videoId: {
                        value: "K-eDe8GDMUo"
                      },
                      videoLazyLoad: {
                        value: false
                      },
                      videoWidth: {
                        value: 600
                      }
                    },
                    templateId: "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
                    templateName: "YouTube Video"
                  },
                  secondaryMediaItem: {
                    id: "1fbaaf0c-1a06-4d0b-b21f-2d4f2f27f764",
                    url: "/Data/Elements/YouTube-Video",
                    name: "YouTube Video",
                    displayName: "YouTube Video",
                    fields: {
                      youTubeShowControls: {
                        value: true
                      },
                      youTubeAutoLoop: {
                        value: true
                      },
                      youTubeAutoPlay: {
                        value: false
                      },
                      youTubeDisableKeyboard: {
                        value: true
                      },
                      youTubeModestBranding: {
                        value: true
                      },
                      youTubeMute: {
                        value: false
                      },
                      videoHeight: {
                        value: 338
                      },
                      videoId: {
                        value: "K-eDe8GDMUo"
                      },
                      videoLazyLoad: {
                        value: false
                      },
                      videoWidth: {
                        value: 600
                      }
                    },
                    templateId: "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
                    templateName: "YouTube Video"
                  },
                  topCopy: {
                    value: "<span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. </span>"
                  },
                  bottomCopy: {
                    value: "<p><span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</span></p>\n<p><span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?</span></p>"
                  },
                  componentSpacing: null,
                  sectionId: {
                    value: ""
                  },
                  theme: {
                    value: ""
                  },
                  headlineLevel: {
                    id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                    name: "Heading 2",
                    displayName: "Heading 2",
                    fields: {
                      Value: {
                        value: "h2"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  headlineText: {
                    value: "Content Block with Media Test"
                  },

                  eventName: {
                    value: ""
                  },
                  eventType: {
                    value: ""
                  },
                  eventZone: {
                    value: ""
                  }
                }
              }
            ]
          }
        },
        {
          id: "6bed4bde-b050-4dce-9360-87cbd18da912",
          fields: {
            tabAnchor: {
              value: ""
            },
            headlineLevel: {
              id: "7f10288a-beb1-4d95-a139-32df40094441",
              url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
              name: "Heading 3",
              displayName: "Heading 3",
              fields: {
                Value: {
                  value: "h3"
                }
              },
              templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
              templateName: "Enum"
            },
            headlineText: {
              value: "Tab Item 2"
            },
          },
          placeholders: {
            components: [
              {
                componentName: "ContentBlockWithMedia2",
                uid: "56d2beff-74da-4aef-9051-061ca573176b",
                dataSource: "{56D2BEFF-74DA-4AEF-9051-061CA573176B}",
                fields: {
                  primaryMediaCaption: {
                    value: "This is a caption with a <a href=\"/\">link</a>."
                  },
                  secondaryMediaCaption: {
                    value: "This is a caption with a <a href=\"/\">link</a>."
                  },
                  primaryMediaItem: {
                    id: "1fbaaf0c-1a06-4d0b-b21f-2d4f2f27f764",
                    url: "/Data/Elements/YouTube-Video",
                    name: "YouTube Video",
                    displayName: "YouTube Video",
                    fields: {
                      youTubeShowControls: {
                        value: true
                      },
                      youTubeAutoLoop: {
                        value: true
                      },
                      youTubeAutoPlay: {
                        value: false
                      },
                      youTubeDisableKeyboard: {
                        value: true
                      },
                      youTubeModestBranding: {
                        value: true
                      },
                      youTubeMute: {
                        value: false
                      },
                      videoHeight: {
                        value: 338
                      },
                      videoId: {
                        value: "K-eDe8GDMUo"
                      },
                      videoLazyLoad: {
                        value: false
                      },
                      videoWidth: {
                        value: 600
                      }
                    },
                    templateId: "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
                    templateName: "YouTube Video"
                  },
                  secondaryMediaItem: {
                    id: "1fbaaf0c-1a06-4d0b-b21f-2d4f2f27f764",
                    url: "/Data/Elements/YouTube-Video",
                    name: "YouTube Video",
                    displayName: "YouTube Video",
                    fields: {
                      youTubeShowControls: {
                        value: true
                      },
                      youTubeAutoLoop: {
                        value: true
                      },
                      youTubeAutoPlay: {
                        value: false
                      },
                      youTubeDisableKeyboard: {
                        value: true
                      },
                      youTubeModestBranding: {
                        value: true
                      },
                      youTubeMute: {
                        value: false
                      },
                      videoHeight: {
                        value: 338
                      },
                      videoId: {
                        value: "K-eDe8GDMUo"
                      },
                      videoLazyLoad: {
                        value: false
                      },
                      videoWidth: {
                        value: 600
                      }
                    },
                    templateId: "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
                    templateName: "YouTube Video"
                  },
                  topCopy: {
                    value: "<span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. </span>"
                  },
                  bottomCopy: {
                    value: "<p><span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</span></p>\n<p><span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?</span></p>"
                  },
                  componentSpacing: null,
                  sectionId: {
                    value: ""
                  },
                  theme: {
                    value: ""
                  },
                  headlineLevel: {
                    id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                    name: "Heading 2",
                    displayName: "Heading 2",
                    fields: {
                      Value: {
                        value: "h2"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  headlineText: {
                    value: "Content Block with Media Test"
                  },

                  eventName: {
                    value: ""
                  },
                  eventType: {
                    value: ""
                  },
                  eventZone: {
                    value: ""
                  }
                }
              }
            ]
          }
        },
        {
          id: "6bed4bde-b050-4dce-9360-87cbd18da913",
          fields: {
            tabAnchor: {
              value: ""
            },
            headlineLevel: {
              id: "7f10288a-beb1-4d95-a139-32df40094441",
              url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
              name: "Heading 3",
              displayName: "Heading 3",
              fields: {
                Value: {
                  value: "h3"
                }
              },
              templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
              templateName: "Enum"
            },
            headlineText: {
              value: "Tab Item 3"
            },
          },
          placeholders: {
            components: [
              {
                componentName: "ContentBlockWithMedia",
                uid: "56d2beff-74da-4aef-9051-061ca573176b",
                dataSource: "{56D2BEFF-74DA-4AEF-9051-061CA573176B}",
                fields: {
                  primaryMediaCaption: {
                    value: "This is a caption with a <a href=\"/\">link</a>."
                  },
                  secondaryMediaCaption: {
                    value: "This is a caption with a <a href=\"/\">link</a>."
                  },
                  primaryMediaItem: {
                    id: "1fbaaf0c-1a06-4d0b-b21f-2d4f2f27f764",
                    url: "/Data/Elements/YouTube-Video",
                    name: "YouTube Video",
                    displayName: "YouTube Video",
                    fields: {
                      youTubeShowControls: {
                        value: true
                      },
                      youTubeAutoLoop: {
                        value: true
                      },
                      youTubeAutoPlay: {
                        value: false
                      },
                      youTubeDisableKeyboard: {
                        value: true
                      },
                      youTubeModestBranding: {
                        value: true
                      },
                      youTubeMute: {
                        value: false
                      },
                      videoHeight: {
                        value: 338
                      },
                      videoId: {
                        value: "K-eDe8GDMUo"
                      },
                      videoLazyLoad: {
                        value: false
                      },
                      videoWidth: {
                        value: 600
                      }
                    },
                    templateId: "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
                    templateName: "YouTube Video"
                  },
                  secondaryMediaItem: {
                    id: "1fbaaf0c-1a06-4d0b-b21f-2d4f2f27f764",
                    url: "/Data/Elements/YouTube-Video",
                    name: "YouTube Video",
                    displayName: "YouTube Video",
                    fields: {
                      youTubeShowControls: {
                        value: true
                      },
                      youTubeAutoLoop: {
                        value: true
                      },
                      youTubeAutoPlay: {
                        value: false
                      },
                      youTubeDisableKeyboard: {
                        value: true
                      },
                      youTubeModestBranding: {
                        value: true
                      },
                      youTubeMute: {
                        value: false
                      },
                      videoHeight: {
                        value: 338
                      },
                      videoId: {
                        value: "K-eDe8GDMUo"
                      },
                      videoLazyLoad: {
                        value: false
                      },
                      videoWidth: {
                        value: 600
                      }
                    },
                    templateId: "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
                    templateName: "YouTube Video"
                  },
                  topCopy: {
                    value: "<span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. </span>"
                  },
                  bottomCopy: {
                    value: "<p><span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</span></p>\n<p><span style=\"background-color: #ffffff; text-align: justify; color: #000000;\">Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?</span></p>"
                  },
                  componentSpacing: null,
                  sectionId: {
                    value: ""
                  },
                  theme: {
                    value: ""
                  },
                  headlineLevel: {
                    id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                    name: "Heading 2",
                    displayName: "Heading 2",
                    fields: {
                      Value: {
                        value: "h2"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  headlineText: {
                    value: "Content Block with Media Test"
                  },

                  eventName: {
                    value: ""
                  },
                  eventType: {
                    value: ""
                  },
                  eventZone: {
                    value: ""
                  }
                }
              }
            ]
          }
        }
      ]
    }
  };

export default defaultData;
