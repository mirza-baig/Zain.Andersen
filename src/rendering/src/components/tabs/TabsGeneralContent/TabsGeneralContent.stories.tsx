// Global
import { Story, Meta } from '@storybook/react';
// Local
import TabsGeneralContent, { TabsGeneralContentProps } from './TabsGeneralContent';
import defaultData from './TabsGeneralContent.mock-data';
import threeTabsData from './TabsGeneralContent.mock-data-with-three-tabs';
import fourTabsData from './TabsGeneralContent.mock-data-with-four-tabs';
import fiveTabsData from './TabsGeneralContent.mock-data-with-five-tabs';
import sixTabsData from './TabsGeneralContent.mock-data-with-six-tabs';
import sevenTabsData from './TabsGeneralContent.mock-data-with-seven-tabs';

export default {
  title: 'Tabs/TabsGeneralContent',
  component: TabsGeneralContent,
} as Meta;
const Template: Story<TabsGeneralContentProps> = (props) => <TabsGeneralContent {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;

export const threeTabs = Template.bind({});
threeTabs.args = threeTabsData;

export const fourTabs = Template.bind({});
fourTabs.args = fourTabsData;

export const fiveTabs = Template.bind({});
fiveTabs.args = fiveTabsData;

export const sixTabs = Template.bind({});
sixTabs.args = sixTabsData;

export const sevenTabs = Template.bind({});
sevenTabs.args = sevenTabsData;
