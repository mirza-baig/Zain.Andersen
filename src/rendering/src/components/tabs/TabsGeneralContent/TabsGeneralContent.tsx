// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import {
  ComponentRendering,
  Placeholder,
  withDatasourceCheck,
} from '@sitecore-jss/sitecore-jss-nextjs';
import React, { useEffect, useState, useRef } from 'react';
import { Tabs, TabList, TabPanel, resetIdCounter } from 'react-tabs';
import type { TabProps } from 'react-tabs';
import { useRouter } from 'next/router';

// Components
import { TabsGeneralContentTheme } from './TabsGeneralContent.theme';
import { Headline } from 'src/helpers/Headline';
import classNames from 'classnames';
import { SliderWrapper } from 'src/helpers/SliderWrapper';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';
import { getEnum } from 'lib/utils';
import { hashCode } from 'src/helpers/Component/Component';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';

type TabItemProps = Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsGeneralContent.TabItem &
  ComponentRendering;

type TabHeadingProps = {
  onClick: () => void;
  children: React.ReactNode | React.ReactNode[];
  index: number;
};

type TabLabelView = 'standard' | 'long';

export type TabsGeneralContentProps =
  Feature.EnterpriseWeb.Enterprise.Components.Tabs.TabsGeneralContent.TabsGeneralContent;
const TabsGeneralContent = (props: TabsGeneralContentProps): JSX.Element => {
  resetIdCounter();
  const tabLabelView = getEnum<TabLabelView>(props.fields?.tabLabelView) || 'standard';
  const numberOfTabs = props.fields?.children.length;
  const router = useRouter();

  const componentRef = useRef<HTMLDivElement>(null);
  const getHashAnchor = (): number => {
    let hashedTabIndex = -1;
    if (typeof window !== 'undefined' && window.location.hash !== '') {
      const hash = window.location.hash.replace('#', '');
      hashedTabIndex = props.fields?.children.findIndex(
        (
          child: Item &
            Feature.EnterpriseWeb.Enterprise.Components.Hero.HeroProductTabs.HeroProductTabItem
        ) => {
          return (
            child.id.localeCompare(hash, undefined, { sensitivity: 'base' }) === 0 ||
            child.fields?.contentId?.value?.localeCompare(hash, undefined, {
              sensitivity: 'base',
            }) === 0
          );
        }
      );
    }
    return hashedTabIndex;
  };

  let defaultTab = 0;
  const hashedTab = getHashAnchor();
  if (hashedTab > -1) {
    defaultTab = hashedTab;
  } else if (!!props.fields?.defaultActiveTab) {
    const defaultTabIndex = props.fields?.children.findIndex((child: Item) => {
      return child.id === props.fields?.defaultActiveTab?.id;
    });
    defaultTab = defaultTabIndex > 0 ? defaultTabIndex : defaultTab;
  }

  const { screenType, currentScreenWidth } = useCurrentScreenType();

  useEffect(() => {
    if (getHashAnchor() > -1 && componentRef.current) {
      componentRef.current.scrollIntoView();
    }

    window.onpopstate = () => {
      const currentHash = getHashAnchor();
      if (currentHash > -1) {
        handleTabIndexChange(getHashAnchor());
      } else {
        router.push(window.location);
      }
    };

    // router is NextRouter module;
    // "getHashAnchor" and "handleTabIndexChange" are functions, so we can ignore react-hooks/exhaustive-deps warning for this suggested dependencies.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const { themeData } = useTheme(
    TabsGeneralContentTheme(screenType ?? 'sm', numberOfTabs, tabLabelView)
  );

  let enoughTabsToScroll = false;
  let slidesToShow = 2;
  let initialSlide = 0;
  let isLongTabLabelView = false;
  if (screenType) {
    if (currentScreenWidth < getBreakpoint('md') && (numberOfTabs > 2 || tabLabelView === 'long')) {
      enoughTabsToScroll = true;
      if (tabLabelView === 'long') {
        slidesToShow = 1;
        initialSlide = defaultTab;
        isLongTabLabelView = true;
      } else {
        slidesToShow = 2;
      }
    } else if (
      currentScreenWidth >= getBreakpoint('md') &&
      currentScreenWidth < getBreakpoint('ml')
    ) {
      slidesToShow = 4;
    } else if (currentScreenWidth >= getBreakpoint('ml')) {
      slidesToShow = 6;
    }
    enoughTabsToScroll = numberOfTabs > slidesToShow ? true : false;
    initialSlide =
      defaultTab >= slidesToShow && !isLongTabLabelView
        ? defaultTab + 1 - slidesToShow
        : initialSlide;
  }

  const handleTabIndexChange = (index: number) => {
    setTabIndex(index);
    const tab = props.fields?.children[index];
    if (!!tab) {
      window.location.hash = tab.fields?.contentId?.value || tab.id;
    }
  };

  const handleSlideIndexChange = (index: number) => {
    if (isLongTabLabelView) {
      handleTabIndexChange(index);
    }
  };

  const sliderSettings = {
    infinite: false,
    speed: 500,
    initialSlide: initialSlide,
    slidesToShow: numberOfTabs > slidesToShow ? slidesToShow : 2,
    slidesToScroll: 1,
    enableNumberedPagination: currentScreenWidth < getBreakpoint('md') && tabLabelView === 'long',
    dots: currentScreenWidth < getBreakpoint('md') && tabLabelView === 'long',
    numberedPaginationClasses: themeData.classes.numberedPaginationClass,
    nextArrowClasses: themeData.classes.nextArrowClass,
    prevArrowClasses: themeData.classes.prevArrowClass,
    arrows: enoughTabsToScroll,
    variableWidth: true,
    className: 'tabsgeneralcontent',
    afterIndexChange: handleSlideIndexChange,
  };

  const TabHeading = ({ onClick, children, index, ...tabProps }: TabHeadingProps) => {
    return (
      <div className={themeData.classes.tabs.outerTabHeadingClass}>
        <div
          className={classNames(
            themeData.classes.tabs.commonTabHeadingClass,
            index === tabIndex
              ? themeData.classes.tabs.selectedTabHeadingClass
              : themeData.classes.tabs.unselectedTabHeadingClass,
            index === props.fields.children.length - 1 ? themeData.classes.tabs.lastTab : ''
          )}
          onClick={onClick}
          onKeyDown={(e: { code: string }) => {
            if (e.code === 'Enter' || e.code === 'Space') {
              onClick();
            }
          }}
          tabIndex={0}
          id={(tabProps as TabProps)?.id || `react-tabs-${index}`}
        >
          {children}
        </div>
      </div>
    );
  };
  TabHeading.tabsRole = 'Tab';

  const [tabIndex, setTabIndex] = useState(defaultTab);

  return (
    <section
      data-component="tabs/tabsgeneralcontent"
      id={props.fields?.sectionId?.value || `id${hashCode(props.rendering?.dataSource)}`}
    >
      <div className={themeData.classes.componentClass} ref={componentRef}>
        <Headline classes={themeData.classes.headlineClass} defaultTag="h2" {...props} />
        {screenType && (
          <Tabs selectedIndex={tabIndex} onSelect={(index: number) => setTabIndex(index)}>
            <TabList className={themeData.classes.tabs.tabListClass}>
              <div role="tab">
                <SliderWrapper sliderSettings={sliderSettings}>
                  {props.fields?.children.map((tab: TabHeadingProps, index: number) => {
                    return (
                      <TabHeading
                        key={index}
                        index={index}
                        onClick={() => {
                          handleTabIndexChange(index);
                        }}
                      >
                        <Headline
                          useTag="h3"
                          classes={classNames(
                            themeData.classes.tabs.headlineClass,
                            index === tabIndex ? themeData.classes.tabs.selectedHeadlineClass : ''
                          )}
                          {...tab}
                        />
                      </TabHeading>
                    );
                  })}
                </SliderWrapper>
              </div>
            </TabList>
            {props.fields?.children.map((tab: TabItemProps, index: number) => {
              return (
                <TabPanel forceRender={true} key={index}>
                  <Placeholder
                    name="components"
                    rendering={tab}
                    forceUpdate={index == tabIndex}
                    render={(tabs) =>
                      tabs.map((tab, index) => (
                        <div key={index} className={classNames('col-span-12')}>
                          {tab}
                        </div>
                      ))
                    }
                    favoriteProducts={props.favoriteProducts}
                  />
                </TabPanel>
              );
            })}
          </Tabs>
        )}
      </div>
    </section>
  );
};

export default withDatasourceCheck()<TabsGeneralContentProps>(TabsGeneralContent);
