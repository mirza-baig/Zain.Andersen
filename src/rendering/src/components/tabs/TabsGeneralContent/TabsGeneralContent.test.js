// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import TabsGeneralContent from './TabsGeneralContent';
import defaultData from './TabsGeneralContent.mock-data';

it('renders correctly', async () => {
  const component = snapshot(TabsGeneralContent, { componentProps: defaultData });
  await hasDataComponent(component, 'tabs/tabsgeneralcontent');
});
