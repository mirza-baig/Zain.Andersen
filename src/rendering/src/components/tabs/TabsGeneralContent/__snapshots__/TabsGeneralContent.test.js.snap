// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="tabs/tabsgeneralcontent"
  id="id1695242607"
>
  <div
    class="theme-white col-span-12 my-8"
  >
    <div
      class="items-top font-sans [&_a:hover]:underline text-s lg:text-m font-heavy pb-s px-m md:max-w-screen-lg lg:mx-auto"
    >
      <h2>
        Tabs General Content Test
      </h2>
    </div>
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <ul
        class="react-tabs__tab-list px-m md:max-w-screen-lg lg:mx-auto"
        role="tablist"
      >
        <div
          role="tab"
        >
          <div
            class="slick-slider tabsgeneralcontent slick-initialized"
          >
            
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="border-r-2 border-r-gray h-full"
                    >
                      <div
                        class="react-tabs__tab flex h-full justify-center text-center items-center pb-s pr-s pl-s pt-m border-t-8 w-auto text-gray border-t-light-gray bg-light-gray hover:border-t-primary hover:bg-white hover:text-black"
                        id="react-tabs-0"
                        tabindex="0"
                      >
                        <div
                          class="items-top font-sans [&_a:hover]:underline font-heavy mb-xxs text-base lg:text-xs justify-center text-center"
                        >
                          <h3>
                            Tab Item 1
                          </h3>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active"
                  data-index="1"
                  style="outline: none;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="border-r-2 border-r-gray h-full"
                    >
                      <div
                        class="react-tabs__tab flex h-full justify-center text-center items-center pb-s pr-s pl-s pt-m border-t-8 w-auto react-tabs__tab--selected border-t-primary border-r-gray border-r-0"
                        id="react-tabs-2"
                        tabindex="0"
                      >
                        <div
                          class="items-top font-sans [&_a:hover]:underline font-heavy mb-xxs text-base lg:text-xs justify-center text-center"
                        >
                          <h3>
                            Tab Item 2
                          </h3>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            
            
          </div>
        </div>
      </ul>
      <div
        aria-labelledby="react-tabs-0"
        class="react-tabs__tab-panel"
        id="react-tabs-1"
        role="tabpanel"
      >
        <div
          class="col-span-12"
        >
          <div
            style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
          >
            <h2>
              ContentBlockWithMedia
            </h2>
            <p>
              JSS component is missing React implementation. See the developer console for more information.
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="react-tabs-2"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        id="react-tabs-3"
        role="tabpanel"
      >
        <div
          class="col-span-12"
        >
          <div
            style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
          >
            <h2>
              ContentBlockWithMedia
            </h2>
            <p>
              JSS component is missing React implementation. See the developer console for more information.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
